mathics/core/expression.py

```diff
@@ -1165,6 +1250,132 @@ def sameQ(self, other: BaseElement) -> bool:
             for element, oelement in zip(self._elements, other.get_elements())
         )
 
+    def sequences(self):
+        cache = self._cache
+        if cache:
+            seq = cache.sequences
+            if seq is not None:
+                return seq
+
+        return self._rebuild_cache().sequences
+
+    def set_head(self, head):
+        self._head = head
+        self._cache = None
+
+    def set_element(self, index: int, value):
```

Comment:
And this is used in 2 places:
- algorithms/parts.py
- core/subexpression.py

Reply:
Maybe it would be useful to add this as comments in the code.

Reply:
@mmatera Rocky already fixed this.

Reply:
Someone feel free to do. In a sense I am always for having comments. But I am trying to understand what why this would be helpful. Set element is setting an element so I would imagine something involving subexpressions and parts to want to modify this. 

If the only purpose is nothing this fact which I hope some sort of automated system would handle then we have the problem with this being wrong when the code changes. Because when someone adds another location or removes one of those locations who is going to know to update the comment. And then it will be wrong. 

So again what are we going to do with this information? You'll notice that in the PR I did add a docstring.